clusterloader2/pkg/state/resource_state.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+)
+
+// ResourceIdentifier is a unique identifier for resource
+type ResourceIdentifier struct {
+	ObjectKind string
+	ApiGroup   string
+}
+
+// resourcesVersionsState represents current version of all used resources.
+type resourcesVersionsState struct {
+	lock              sync.RWMutex
+	resourcesVersions map[ResourceIdentifier]string
+}
+
+// newResourcesVersionState creates new resources versions state.
+func newResourcesVersionsState() *resourcesVersionsState {
+	return &resourcesVersionsState{
+		resourcesVersions: make(map[ResourceIdentifier]string),
+	}
+}
+
+// Get returns state of current resource version.
+func (rs *resourcesVersionsState) Get(identifier ResourceIdentifier) (string, bool) {
+	rs.lock.RLock()
+	defer rs.lock.RUnlock()
+	version, exists := rs.resourcesVersions[identifier]
+	return version, exists
+}
+
+// Set stores information about current resource version.
+func (rs *resourcesVersionsState) Set(identifier ResourceIdentifier, version string) {
+	rs.lock.Lock()
+	defer rs.lock.Unlock()
+	rs.resourcesVersions[identifier] = version
+}
+
+// Delete removes information about given resource type.
+func (rs *resourcesVersionsState) Delete(identifier ResourceIdentifier) error {
```

Comment:
I don't think that's useful - I don't see a way how this could be used :-)

Reply:
Function removed.